superset-frontend/webpack.config.js

```diff
@@ -199,57 +189,29 @@ const config = {
       {
         test: /\.tsx?$/,
         use: [
-          { loader: 'cache-loader' },
-          {
-            loader: 'thread-loader',
-            options: {
-              // there should be 1 cpu for the fork-ts-checker-webpack-plugin
-              workers: os.cpus().length - 1,
-            },
-          },
+          'thread-loader',
+          babelLoader,
           {
             loader: 'ts-loader',
             options: {
               // transpile only in happyPack mode
               // type checking is done via fork-ts-checker-webpack-plugin
               happyPackMode: true,
+              transpileOnly: true,
             },
           },
         ],
       },
       {
         test: /\.jsx?$/,
-        exclude: /node_modules/,
-        include: APP_DIR,
-        loader: 'babel-loader',
-      },
-      {
-        // handle symlinked modules
-        // for debugging @superset-ui packages via npm link
-        test: /\.jsx?$/,
-        include: /node_modules\/[@]superset[-]ui.+\/src/,
-        use: [
-          {
-            loader: 'babel-loader',
-            options: BABEL_JAVASCRIPT_OPTIONS,
-          },
-        ],
-      },
-      {
-        // handle symlinked modules
-        // for debugging @superset-ui packages via npm link
-        test: /\.tsx?$/,
-        include: /node_modules\/[@]superset[-]ui.+\/src/,
-        use: [
-          {
-            loader: 'babel-loader',
-            options: BABEL_TYPESCRIPT_OPTIONS,
-          },
-        ],
+        // include source code for plugins, but exclude node_modules within them
+        exclude: [/superset-ui.*\/node_modules\/.*/],
+        include: [new RegExp(`${APP_DIR}/src`), /superset-ui.*\/src/],
+        use: [babelLoader],
```

Comment:
We should allow any `superset-ui` packages, in case someone wants to publish their own plugin but not under the `@superset-ui` org.